libs/ccc/coef/impl.py

```diff
@@ -515,6 +517,32 @@ def cdist_func(x, y):
     return max_ari_list, max_part_idx_list, pvalues
 
 
+def get_n_workers(n_jobs: int | None) -> int:
+    """
+    Helper function to get the number of workers for parallel processing.
+
+    Args:
+        n_jobs: value specified by the main ccc function.
+    Returns:
+        The number of workers to use for parallel processing
+    """
+    n_cpu_cores = os.cpu_count()
+    if n_cpu_cores is None:
+        raise ValueError("Could not determine the number of CPU cores. Please specify a positive value of n_jobs")
+
+    n_workers = n_cpu_cores
+    if n_jobs is None:
+        return n_workers
+
+    n_workers = os.cpu_count() + n_jobs if n_jobs < 0 else n_jobs
+
+    if n_workers < 1:
+        raise ValueError(f"The number of threads/processes to use must be greater than 0. Got {n_workers}."
+                         "Please check the n_jobs argument provided")
```

Comment:
Looks good. If a user specifies more threads than actual CPU cores, that's a problem for the user (using more resources than actual ones) and we should not check for that.

Reply:
Yep! As `ThreadPoolExecutor(max_workers=n_workers)` also only complains when n_workers < 0. We don't need to check for that case